### PR TITLE
Raw: Menu Extension

### PR DESCRIPTION
**Pull request by RandyStevens**
### Description
This changeset adds `5/OCS-N`, which defines an extension to the core _OpenCannabis Specification_ to provide structures that express menus of concrete products.

### Changelog
- [x] Early menu extension proposal
- [x] All menu docs and structures moved over
- [x] All menu structures documented
- [x] Review
  - [x] Retail WG (@OpenCannabis/retail)
  - [x] Technology WG (@OpenCannabis/technology)

### CODE REVIEW

File: 5/README.md [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+| cartridges | [StaticMenu.CartridgesEntry](#opencannabis.products.menu.StaticMenu.CartridgesEntry) | repeated | Specifies CARTRIDGE products attached to this menu. |
+| edibles | [StaticMenu.EdiblesEntry](#opencannabis.products.menu.StaticMenu.EdiblesEntry) | repeated | Specifies EDIBLE products attached to this menu. |
+| extracts | [StaticMenu.ExtractsEntry](#opencannabis.products.menu.StaticMenu.ExtractsEntry) | repeated | Specifies EXTRACT products attached to this menu. |
+| flowers | [StaticMenu.FlowersEntry](#opencannabis.products.menu.StaticMenu.FlowersEntry) | repeated | Specifies FLOWER products attached to this menu. |
+| merchandise | [StaticMenu.MerchandiseEntry](#opencannabis.products.menu.StaticMenu.MerchandiseEntry) | repeated | Specifies MERCHANDISE products attached to this menu. |
+| plants | [StaticMenu.PlantsEntry](#opencannabis.products.menu.StaticMenu.PlantsEntry) | repeated | Specifies PLANT products attached to this menu. |
+| prerolls | [StaticMenu.PrerollsEntry](#opencannabis.products.menu.StaticMenu.PrerollsEntry) | repeated | Specifies PREROLL products attached to this menu. |
+
```
Comment (RandyStevens) [inline 5/README.md:257]: Static menu descriptions for the majority of buisnesses

File: 5/README.md [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+| key | [opencannabis.base.ProductKey](#opencannabis.base.ProductKey) |  | Section that this data is attached to. |
+| apothecary | [opencannabis.products.Apothecary](#opencannabis.products.Apothecary) |  | Apothecary product. |
+| cartridge | [opencannabis.products.Cartridge](#opencannabis.products.Cartridge) |  | Cartridge product. |
+| edible | [opencannabis.products.Edible](#opencannabis.products.Edible) |  | Edible product. |
+| extract | [opencannabis.products.Extract](#opencannabis.products.Extract) |  | Extract product. |
+| flower | [opencannabis.products.Flower](#opencannabis.products.Flower) |  | Flower product. |
+| merchandise | [opencannabis.products.Merchandise](#opencannabis.products.Merchandise) |  | Merchandise product. |
+| plant | [opencannabis.products.Plant](#opencannabis.products.Plant) |  | Plant product. |
+| preroll | [opencannabis.products.Preroll](#opencannabis.products.Preroll) |  | Preroll product. |
+
```
Comment (RandyStevens) [inline 5/README.md]: Product descriptions for each section

File: 5/README.md [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+| ----- | ---- | ----- | ----------- |
+| key | [string](#string) |  |  |
+| value | [opencannabis.products.Preroll](#opencannabis.products.Preroll) |  |  |
+
+
+<a name="opencannabis.products.menu.Flag"/>
+
+### Flag
+Enumerates flags that can be set on a menu.
+
```
Comment (RandyStevens) [inline 5/README.md:348]: Specific Flags for a menu, I.E. if you have products you may not display so you have a "draft" version of your menu.

File: 3/3-Core-Structures.md [hunk 4]
```diff
@@ -58,6 +58,30 @@ separately in [Part 1.2, Extension Structures](4-Extension-Structures.md).
...
+    - [ApothecaryType](#opencannabis.products.ApothecaryType)
+    - [Flower](#opencannabis.products.Flower)
+    - [Edible](#opencannabis.products.Edible)
+    - [EdibleIngredient](#opencannabis.products.EdibleIngredient)
+    - [EdibleFlag](#opencannabis.products.EdibleFlag)
+    - [EdibleType](#opencannabis.products.EdibleType)
+    - [Extract](#opencannabis.products.Extract)
+    - [ExtractFlag](#opencannabis.products.ExtractFlag)
+    - [ExtractType](#opencannabis.products.ExtractType)
+    - [DistributionPolicy](#opencannabis.products.distribution.DistributionPolicy)
```
Comment (RandyStevens) [inline 3/3-Core-Structures.md:74]: Did a little revision to the Table of Contents in base-models

File: 5/README.md (hunk 2 above)
Comment (sgammon) [reply 5/README.md]: hm, these should definitely go in the core spec. the menus are composed of these objects, but these object themselves shouldn't be specified in an extension.

**Review by sgammon — CHANGES_REQUESTED** [review]
Let's move the concrete product models (i.e. `Flower`, `Edible`, etc.) into the core spec.

**Comment by RandyStevens** [comment]
Ok i think i managed. Ready for another Review

File: 3/4-Extension-Structures.md [hunk 5]
```diff
@@ -10,12 +10,13 @@ Definitions considered core to the main specification are enumerated separately
...
-- `oauth`: OAuth2 implementation structures.
-- `commerce`: Commercial ordering and fulfillment structures.
-- `accounting`: Bookkeeping, accounting, taxes, and so on.
+- [5/OCS-N](../5/README.md) / `menu` : Distribution channels, product labels, product types, etc.
+- [6/OCS-C](../6/README.md) / `commerce`: Commercial ordering and fulfillment structures.
+- [7/OCS-L](../7/README.md) / `labtesting`: Laboratory testing and QA.
+- [8/OCS-P](../8/README.md) / `pricing`: Pricing schemes and structures.
+- [9/OCS-X](../9/README.md) / `proximity`: Bluetooth and GPS-related structures.
+- [10/OCS-O](../10/README.md) / `oauth`: OAuth2 implementation structures.
+- [11/OCS-A](../11/README.md) / `accounting`: Bookkeeping, accounting, taxes, and so on.
```
Comment (sgammon) [inline 3/4-Extension-Structures.md:19]: nice, although we may need to change these IDs later

**Review by sgammon — APPROVED** [review]

File: 3/4-Extension-Structures.md (hunk 5 above)
Comment (RandyStevens) [reply 3/4-Extension-Structures.md:19]: Thats fine, I tried to leave them unique and understandable, but jumping from P to X to A is a little weird when you look at it

**Pull request merged**